### DISCUSSION TITLE
Setting client_max_body_size via docker compose

### DISCUSSION THREAD

**Discussion by kfunkeweb** [nginx-proxy/nginx-proxy#2482] [Ideas]
Hi,
I really love this project because it just works without needing to do any configuration. However, the most common customization I need is setting client_max_body_size to a higher value. Otherwise uploads don't work in a lot of software. I currently do this by editing the vhost file, but I would prefer to be able to change it directly in my compose file, so I don't have to manually do anything in the docker volumes.

**Comment by strichte** [comment]
I second this. The nginx default for client_max_body_size  is 1M. Wordpress is a prominent example where this is not enough. Most theme and plugin uploads will require a lot more than that.

I currently solve this by adding a client_max_body_size in the server section of the nginx.tmpl file. This will give every server the same client_max_body_size , which is not ideal. Best option would be an env variable in each server section, similar to LETSENCRYPT_HOST, i.e. PROXY_MAX_BODY_SIZE.

At a minimum I would suggest to add a note in the documentation, so users are aware of this.

**Comment by jaxor24** [comment]
Just ran into this issue. Agree it would be great to configure per server.

**Comment by MPZinke** [comment]
Adding my request for this to the pool. Thank you for all the hard work you have done.
